### PR TITLE
[OpenVINO-EP 2021.4] Add/update Dockerfiles w.r.t OpenVINO 2021.4 Version

### PR DESCRIPTION
**Pull request by MaajidKhan**
**Description:**
-> Rearchitected OpenVINO EP ubuntu python dockerfile using multi-stage dockerfile approach to reduce pre-built docker image size.
-> Uploaded OpenVINO EP Dockerfile for CentOS7 with Python API
-> updated c# dockerfile w.r.t OpenVINO 2021.4
-> Updated documentation with location of pre-built Docker image released by Intel

**Motivation and Context:**
Intel will start releasing pre-built docker image for ONNX RT with OpenVINO EP in Dockerhub. Above changes help this objective.

### CODE REVIEW

File: dockerfiles/README.md [hunk 1]
```diff
@@ -190,8 +188,7 @@ If the *device_type* runtime config option is not explicitly specified, CPU will
     - Note that if OpenVINO was installed with root permissions, this file has to be changed with the same permissions.
 5. Run the docker image by mounting the device drivers
     ```
-    docker run -it --rm --device-cgroup-rule='c 189:* rmw' -v /dev/bus/usb:/dev/bus/usb --mount type=bind,source=/var/tmp,destination=/var/tmp --device /dev/ion:/dev/ion  onnxruntime-vadm:latest
-
+    docker run -itu root:root --rm --device-cgroup-rule='c 189:* rmw' -v /dev/bus/usb:/dev/bus/usb --mount type=bind,source=/var/tmp,destination=/var/tmp --device /dev/ion:/dev/ion  onnxruntime-vadm:latest
```
Comment (oliviajain) [inline dockerfiles/README.md:191]: Are cgroup rules only needed for vadm? What about Myriad_fp16?

**Comment by oliviajain** [comment]
Validated ORT + CentOS for CPU_FP32.

File: dockerfiles/README.md (hunk 1 above)
Comment (MaajidKhan) [reply dockerfiles/README.md:191]: Yes cgroup rule is required for Myriad_fp16 as well and this is already documented here.

**3. Run the docker image by mounting the device drivers
    ```
    docker run -it --rm --device-cgroup-rule='c 189:* rmw' -v /dev/bus/usb:/dev/bus/usb onnxruntime-myriad:latest
    ```**
  
Refer here:
https://github.com/microsoft/onnxruntime/blob/c08433857ba7e67be3e67ea630326e0bf31f5e76/dockerfiles/README.md#openvino-on-myriad-vpu-accelerator

File: dockerfiles/Dockerfile.openvino-centos7 [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    wget -O intel-opencl-19.41.14441-1.el7.x86_64.rpm https://sourceforge.net/projects/intel-compute-runtime/files/19.41.14441/centos-7/intel-opencl-19.41.14441-1.el7.x86_64.rpm/download && \
+    wget -O intel-igc-opencl-devel-1.0.2597-1.el7.x86_64.rpm https://sourceforge.net/projects/intel-compute-runtime/files/19.41.14441/centos-7/intel-igc-opencl-devel-1.0.2597-1.el7.x86_64.rpm/download && \
+    wget -O intel-igc-opencl-1.0.2597-1.el7.x86_64.rpm https://sourceforge.net/projects/intel-compute-runtime/files/19.41.14441/centos-7/intel-igc-opencl-1.0.2597-1.el7.x86_64.rpm/download && \
+    wget -O intel-gmmlib-19.3.2-1.el7.x86_64.rpm https://sourceforge.net/projects/intel-compute-runtime/files/19.41.14441/centos-7/intel-gmmlib-19.3.2-1.el7.x86_64.rpm/download && \
+    wget -O intel-gmmlib-devel-19.3.2-1.el7.x86_64.rpm https://sourceforge.net/projects/intel-compute-runtime/files/19.41.14441/centos-7/intel-gmmlib-devel-19.3.2-1.el7.x86_64.rpm/download && \
+    rpm -i /tmp/opencl/*.rpm  && \
+    ldconfig  && \
+    rm -rf /tmp/opencl && \
+# Installing gcc-7
```
Comment (snnn) [inline dockerfiles/Dockerfile.openvino-centos7:81]: Could you use gcc-10 instead? gcc-7 doesn't have full C++17 support.

Comment (MaajidKhan) [reply dockerfiles/Dockerfile.openvino-centos7]: @snnn Updated gcc version to 10.
Also validated the new changes to dockerfile by building a image locally and validating all the hardware's we support - CPU, GPU, MYRIADX, VADM inside the container.

**Comment by oliviajain** [comment]
@MaajidKhan could you resolve the merge conflict for Dockerfile.openvino?

> Pull request contains merge conflicts.

@MaajidKhan could you resolve the merge conflict for Dockerfile.openvino?

**Comment by MaajidKhan** [comment]
> @MaajidKhan could you resolve the merge conflict for Dockerfile.openvino?
> 
> > Pull request contains merge conflicts.

@oliviajain Hello oliva, I just fixed them. can you trigger the pipelines now. 👍

**Review by oliviajain — APPROVED** [review]

**Pull request merged**